packages/compass-data-modeling/src/services/open-and-download-diagram.ts

```diff
@@ -25,3 +27,67 @@ export function getDownloadDiagramContent(name: string, edits: Edit[]) {
     edits: Buffer.from(JSON.stringify(edits)).toString('base64'),
   };
 }
+
+export async function getDiagramContentsFromFile(
+  file: File
+): Promise<{ name: string; edits: [SetModelEdit, ...Edit[]] }> {
+  const reader = new FileReader();
+  return new Promise((resolve, reject) => {
+    reader.onload = (event) => {
+      const content = event.target?.result;
+      if (typeof content !== 'string') {
+        return reject(new Error('Invalid file contents'));
+      }
+      try {
+        const parsedContent = JSON.parse(content);
+
+        if (
+          parsedContent.version !== kCurrentVersion ||
+          parsedContent.type !== kFileTypeDescription
+        ) {
+          return reject(new Error('Unsupported diagram file format'));
```

Comment:
Nit: I think makes more sense to just throw here so that the same error message adjustments can be applied to all errors that parsing the file contents can produce

```suggestion
          new Error('Unsupported diagram file format');
```